--- php-pecl-runkit-0.9/runkit_import.c~	2006-06-07 00:06:08.000000000 +0300
+++ php-pecl-runkit-0.9/runkit_import.c	2008-10-13 01:24:08.073851193 +0300
@@ -227,7 +227,7 @@
 			char *cname = NULL, *pname = key;
 
 #ifdef ZEND_ENGINE_2
-			zend_unmangle_property_name(key, &cname, &pname);
+			zend_unmangle_property_name(key, key_len - 1, &cname, &pname);
 #endif
 			if (zend_hash_exists(&dce->default_properties, key, key_len)) {
 				if (override) {
--- php-pecl-runkit-0.9/runkit_sandbox.c~	2008-10-13 01:36:43.000000000 +0300
+++ php-pecl-runkit-0.9/runkit_sandbox.c	2008-10-13 01:36:51.665946502 +0300
@@ -1467,8 +1467,7 @@
 	zval *retval; \
 \
 	ALLOC_ZVAL(retval); \
-	Z_TYPE_P(retval) = IS_BOOL; \
-	Z_BVAL_P(retval) = objval->name; \
+	ZVAL_BOOL(retval, objval->name); \
 	retval->refcount = 0; \
 	retval->is_ref = 0; \
 \
