--- php-pecl-runkit-0.9/runkit.c~	2010-06-06 18:30:08.000000000 +0300
+++ php-pecl-runkit-0.9/runkit.c	2010-06-06 18:30:11.272865268 +0300
@@ -195,11 +195,7 @@
  */
 PHP_MINIT_FUNCTION(runkit)
 {
-#ifdef ZTS
-	ts_allocate_id(&runkit_globals_id, sizeof(zend_runkit_globals), php_runkit_globals_ctor, NULL);
-#else
-	php_runkit_globals_ctor(&runkit_globals);
-#endif
+	ZEND_INIT_MODULE_GLOBALS(runkit, php_runkit_globals_ctor, php_runkit_globals_ctor);
 
 #if defined(PHP_RUNKIT_SUPERGLOBALS) || defined(PHP_RUNKIT_MANIPULATION)
 	REGISTER_INI_ENTRIES();
@@ -364,7 +360,7 @@
 {
 #ifdef PHP_RUNKIT_SUPERGLOBALS
 	if (RUNKIT_G(superglobals)) {
-		zend_hash_apply(RUNKIT_G(superglobals), php_runkit_superglobal_dtor TSRMLS_CC);
+		zend_hash_apply(RUNKIT_G(superglobals), (apply_func_t) php_runkit_superglobal_dtor TSRMLS_CC);
 
 		zend_hash_destroy(RUNKIT_G(superglobals));
 		FREE_HASHTABLE(RUNKIT_G(superglobals));
@@ -374,7 +370,7 @@
 #ifdef PHP_RUNKIT_MANIPULATION
 	if (RUNKIT_G(misplaced_internal_functions)) {
 		/* Just wipe out rename-to targets before restoring originals */
-		zend_hash_apply(RUNKIT_G(misplaced_internal_functions), php_runkit_destroy_misplaced_functions TSRMLS_CC);
+		zend_hash_apply(RUNKIT_G(misplaced_internal_functions), (apply_func_t) php_runkit_destroy_misplaced_functions TSRMLS_CC);
 		zend_hash_destroy(RUNKIT_G(misplaced_internal_functions));
 		FREE_HASHTABLE(RUNKIT_G(misplaced_internal_functions));
 		RUNKIT_G(misplaced_internal_functions) = NULL;
@@ -382,7 +378,7 @@
 
 	if (RUNKIT_G(replaced_internal_functions)) {
 		/* Restore internal functions */
-		zend_hash_apply_with_arguments(RUNKIT_G(replaced_internal_functions), php_runkit_restore_internal_functions, 1, RUNKIT_TSRMLS_C);
+		zend_hash_apply_with_arguments(RUNKIT_G(replaced_internal_functions), (apply_func_args_t) php_runkit_restore_internal_functions, 1, RUNKIT_TSRMLS_C);
 		zend_hash_destroy(RUNKIT_G(replaced_internal_functions));
 		FREE_HASHTABLE(RUNKIT_G(replaced_internal_functions));
 		RUNKIT_G(replaced_internal_functions) = NULL;
--- php-pecl-runkit-0.9/php_runkit.h~	2010-06-06 18:31:09.000000000 +0300
+++ php-pecl-runkit-0.9/php_runkit.h	2010-06-06 19:37:40.356209521 +0300
@@ -173,7 +173,7 @@
 #define PHP_RUNKIT_DECL_STRING_PARAM(p)			char *p; int p##_len;
 #define PHP_RUNKIT_STRING_SPEC					"s"
 #define PHP_RUNKIT_STRING_PARAM(p)				&p, &p##_len
-#define PHP_RUNKIT_STRTOLOWER(p)				php_strtolower(&p, &p##_len)
+#define PHP_RUNKIT_STRTOLOWER(p)				php_strtolower(p, p##_len)
 #define PHP_RUNKIT_STRING_LEN(param,addtl)		(param##_len + (addtl))
 #define PHP_RUNKIT_STRING_TYPE(param)			IS_STRING
 #define PHP_RUNKIT_HASH_FIND(hash,param,ppvar)	zend_hash_find(hash, param, param##_len + 1, (void**)ppvar)
--- php-pecl-runkit-0.9/runkit_functions.c~	2010-06-06 19:42:18.000000000 +0300
+++ php-pecl-runkit-0.9/runkit_functions.c	2010-06-06 19:42:20.332432692 +0300
@@ -95,7 +95,7 @@
 
 			if (!RUNKIT_G(misplaced_internal_functions)) {
 				ALLOC_HASHTABLE(RUNKIT_G(misplaced_internal_functions));
-				zend_hash_init(RUNKIT_G(misplaced_internal_functions), 4, NULL, php_runkit_hash_key_dtor, 0);
+				zend_hash_init(RUNKIT_G(misplaced_internal_functions), 4, NULL, (dtor_func_t) php_runkit_hash_key_dtor, 0);
 			}
 			hash_key.nKeyLength = fname_len + 1;
 			PHP_RUNKIT_HASH_KEY(&hash_key) = estrndup(fname, PHP_RUNKIT_HASH_KEYLEN(&hash_key));
@@ -468,7 +468,7 @@
 		PHP_RUNKIT_HASH_KEY(&hash_key) = estrndup(dfunc, PHP_RUNKIT_HASH_KEYLEN(&hash_key));
 		if (!RUNKIT_G(misplaced_internal_functions)) {
 			ALLOC_HASHTABLE(RUNKIT_G(misplaced_internal_functions));
-			zend_hash_init(RUNKIT_G(misplaced_internal_functions), 4, NULL, php_runkit_hash_key_dtor, 0);
+			zend_hash_init(RUNKIT_G(misplaced_internal_functions), 4, NULL, (dtor_func_t)php_runkit_hash_key_dtor, 0);
 		}
 		zend_hash_next_index_insert(RUNKIT_G(misplaced_internal_functions), (void*)&hash_key, sizeof(zend_hash_key), NULL);
 	}
--- php-pecl-runkit-0.9/runkit_sandbox.c~	2010-06-06 19:53:07.000000000 +0300
+++ php-pecl-runkit-0.9/runkit_sandbox.c	2010-06-06 19:53:09.195711797 +0300
@@ -1570,7 +1570,7 @@
 	char *name;
 	int name_len;
 	zval *(*getter)(php_runkit_sandbox_object *objval TSRMLS_DC);
-	int (*setter)(php_runkit_sandbox_object *objval, zval *value TSRMLS_DC);
+	void (*setter)(php_runkit_sandbox_object *objval, zval *value TSRMLS_DC);
 };
 
 struct _php_runkit_sandbox_settings php_runkit_sandbox_settings[] = {
